include/libdnf/conf/option_binds.hpp

```diff
@@ -1,83 +1,91 @@
+
+    class OptionAlreadyExists : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_name() const noexcept override { return "OptionAlreadyExists"; }
+        const char * get_description() const noexcept override { return "Option already exists"; }
     };
 
+    /// Extends the option with user-defined functions for processing a new value and converting value to a string.
+    /// It is used as additional level of processing when the option is accesed by its text name.
     class Item final {
     public:
-        typedef std::function<void(Option::Priority, const std::string &)> NewStringFunc;
-        typedef std::function<const std::string & ()> GetValueStringFunc;
+        using NewStringFunc = std::function<void(Option::Priority, const std::string &)>;
+        using GetValueStringFunc = std::function<const std::string &()>;
 
-        Option::Priority getPriority() const;
-        void newString(Option::Priority priority, const std::string & value);
-        std::string getValueString() const;
-        bool getAddValue() const;
+        Option::Priority get_priority() const;
+        void new_string(Option::Priority priority, const std::string & value);
+        std::string get_value_string() const;
+        bool get_add_value() const;
 
     private:
         friend class OptionBinds;
 
-        Item(Option & option, const NewStringFunc & newString,
-             const GetValueStringFunc & getValueString, bool addValue);
-        Item(Option & option, NewStringFunc && newString,
-             GetValueStringFunc && getValueString, bool addValue);
-        Item(Option & option);
+        Item(Option & option, NewStringFunc new_string, GetValueStringFunc get_value_string, bool add_value);
+        explicit Item(Option & option);
         Option * option;
-        NewStringFunc newStr;
-        GetValueStringFunc getValueStr;
-        bool addValue{false}; // hint that new value be added
+        NewStringFunc new_str;
+        GetValueStringFunc get_value_str;
+        bool add_value{false};  // hint that new value be added/appended
     };
 
-    typedef std::map<std::string, Item> Container;
-    typedef Container::iterator iterator;
-    typedef Container::const_iterator const_iterator;
+    using Container = std::map<std::string, Item>;
+    using iterator = Container::iterator;
+    using const_iterator = Container::const_iterator;
 
-    Item & add(const std::string & id, Option & option, const Item::NewStringFunc & newString,
-                 const Item::GetValueStringFunc & getValueString, bool addValue);
-    Item & add(const std::string & id, Option & option, Item::NewStringFunc && newString,
-                 Item::GetValueStringFunc && getValueString, bool addValue);
+    Item & add(
+        const std::string & id,
+        Option & option,
+        Item::NewStringFunc new_string,
+        Item::GetValueStringFunc get_value_string,
+        bool add_value);
```

Comment:
I am sorry, please what it does?
`Item::NewStringFunc new_string,
 Item::GetValueStringFunc get_value_string,`
I am not sure if names are correct.

Reply:
@j-mracek 
`Item::NewStringFunc new_string` - custom function that can modify behaviour of `void new_string(Option::Priority priority, const std::string & value);` method. It can modify value that is set to the option. Eg. It's used for "append" options.

`Item::GetValueStringFunc get_value_string` - custom function that can modify/preprocess the value  obtained by `std::string get_value_string() const` method.

I can change the name of arguments:
`new_string` -> `new_string_func` or `new_string_cb`
`get_value_string` -> `get_value_string_func` or `get_value_string_cb`

Reply:
@j-mracek According discussing I renamed they to `new_string_func`, `get_value_string_func`